src/goto-harness/function_call_harness_generator.cpp

```diff
@@ -42,6 +72,22 @@ void function_call_harness_generatort::handle_option(
   {
     p_impl->function = require_exactly_one_value(option, values);
   }
+  else if(option == FUNCTION_HARNESS_GENERATOR_NONDET_GLOBALS_OPT)
+  {
+    p_impl->nondet_globals = true;
+  }
+  else if(option == FUNCTION_HARNESS_GENERATOR_MIN_NULL_TREE_DEPTH_OPT)
+  {
+    auto const value = require_exactly_one_value(option, values);
+    p_impl->recursive_initialization_config.min_null_tree_depth =
+      std::stoul(value);
```

Comment:
Don't we have some facility to extract numbers from string safely via `std::optional`?

Reply:
We probably do, we'll see how we can improve these

Reply:
So, I still think this would be a good idea but we can’t find it. There are the `safe_` conversion functions, but those just assert that the conversion succeeds rather than returning an `optionalt`

Reply:
Well `stoul` is safe: throwing exceptions on failure. May want to catch those.

Reply:
I'd use `safe_string2size_t` here, unless you actually want to throw an exception.

Reply:
@tautschnig Considering this is user input I think throwing an exception here would be appropriate. I don't think asserting that user input is always correct is appropriate.

Reply:
@xbauch @tautschnig I added https://github.com/diffblue/cbmc/pull/4192, which adds a more convenient way to convert strings to integers and handling failures locally.

Reply:
That's great, indeed we should not be failing an assertion when reporting invalid user input.